src/app/debug/debug-tree/debug-tree.component.ts

```diff
@@ -67,6 +70,19 @@ export class DebugTreeComponent implements OnDestroy {
     this.showMultipleAtATimeSubscription.unsubscribe();
   }
 
+  handleError(): (error: HttpErrorResponse) => Observable<any> {
+    return (error: HttpErrorResponse): Observable<any> => {
+      const message = error.error;
+      if (message && message.includes('- detailed error message -')) {
+        const errorMessageParts = message.split('- detailed error message -');
+        this.toastService.showDanger(errorMessageParts[0], errorMessageParts[1]);
+      } else {
+        this.toastService.showDanger(error.message, '');
+      }
+      return of(error);
+    };
+  }
```

Comment:
Move this to its own class, remember the [DRY principle](https://wiki.c2.com/?DontRepeatYourself)

Reply:
I have put it in its own class now